Code/Framework/AzToolsFramework/Tests/Prefab/Benchmark/Propagation/PropagationBenchmarkFixture.cpp

```diff
@@ -0,0 +1,208 @@
+        }
+
+        state.SetComplexityN(state.range());
+    }
+
+    void PropagationBenchmarkFixture::AddComponent(benchmark::State& state)
+    {
+        m_entityModify->Deactivate();
+        for (auto _ : state)
+        {
+            state.PauseTiming();
+            ASSERT_EQ(m_entityModify->GetComponents().size(), 1);
+
+            // Add another component and update the template to capture this change.
+            AZ::Component* inspectorComponent = m_entityModify->CreateComponent<AzToolsFramework::Components::EditorInspectorComponent>();
+            ASSERT_EQ(m_entityModify->GetComponents().size(), 2);
+            UpdateTemplate();
+            state.ResumeTiming();
+
+            m_instanceUpdateExecutorInterface->UpdateTemplateInstancesInQueue();
+            state.PauseTiming();
+
+            // Remove the second component added. This will make sure that when multiple iterations are done, we will always be going from
+            // one components to two components.
+            m_entityModify->RemoveComponent(inspectorComponent);
+            ASSERT_EQ(m_entityModify->GetComponents().size(), 1);
+            delete inspectorComponent;
+            inspectorComponent = nullptr;
+            UpdateTemplate();
+        }
+
+        state.SetComplexityN(state.range());
+    }
+
+    void PropagationBenchmarkFixture::RemoveComponent(benchmark::State& state)
+    {
+        m_entityModify->Deactivate();
+
+        for (auto _ : state)
+        {
+            state.PauseTiming();
+            const AZStd::vector<AZ::Component*>& components = m_entityModify->GetComponents();
+            ASSERT_EQ(m_entityModify->GetComponents().size(), 1);
+            AZ::Component* transformComponent = components.front();
+            m_entityModify->RemoveComponent(transformComponent);
+            delete transformComponent;
+            transformComponent = nullptr;
+            ASSERT_EQ(m_entityModify->GetComponents().size(), 0);
+            UpdateTemplate();
+            state.ResumeTiming();
+
+            m_instanceUpdateExecutorInterface->UpdateTemplateInstancesInQueue();
+            state.PauseTiming();
+
+            // Add the component back. This will make sure that when multiple iterations are done, we will always be going from
+            // zero components to one component.
+            m_entityModify->CreateComponent(AZ::TransformComponentTypeId);
+            ASSERT_EQ(m_entityModify->GetComponents().size(), 1);
+            UpdateTemplate();
+        }
+
+        state.SetComplexityN(state.range());
+    }
+
+    void PropagationBenchmarkFixture::AddEntity(benchmark::State& state)
+    {
+        for (auto _ : state)
+        {
+            state.PauseTiming();
+
+            // Add an entity and update the template.
+            AZStd::unique_ptr<AZ::Entity> newEntity = AZStd::make_unique<AZ::Entity>("Added Entity");
+            AZ::EntityId newEntityId = newEntity->GetId();
+            ASSERT_TRUE(newEntity != nullptr);
```

Comment:
The `ASSERT_*` calls should be removed.
They are for googletest, not for google benchmarks(Also ASSERT_TRUE should only be used for a boolean expression, if a relational operator should not be `ASSERT_TRUE/ASSERT_FALSE`, instead the proper ASSERT_<Relational> macro should be used).

Reply:
I have found the asserts to be useful in validating that the benchmarks are being run on the correct expected data. They will help if any code changes causes some bugs in benchmarks setup.

If ASSERT_ shouldn't be used, can I use EXPECT_  or is that also only intended to be used for google tests ? What other forms of validations are okay for benchmarks?

Reply:
After further discussion offline, removed the macros since they are added to catch any issues with the setup before benchmarsk. Instead, validations using AZ_Error and AZ_Assert will be added to Setup and Teardown in the future as seen necessary.